fighters/rosetta/src/opff.rs

```diff
@@ -12,8 +12,80 @@ unsafe fn launch_star_cancel(boma: &mut BattleObjectModuleAccessor, status_kind:
     }
 }
 
-pub unsafe fn moveset(boma: &mut BattleObjectModuleAccessor, id: usize, cat: [i32 ; 4], status_kind: i32, situation_kind: i32, motion_kind: u64, stick_x: f32, stick_y: f32, facing: f32, frame: f32) {
+extern "Rust" {
+    fn gimmick_flash(boma: &mut BattleObjectModuleAccessor);
+}
+
+//Rosalina Teleport
+unsafe fn teleport(fighter: &mut L2CFighterCommon, boma: &mut BattleObjectModuleAccessor, status_kind: i32) {
+				let fighter_kind = smash::app::utility::get_kind(boma);
+				let entry_id = WorkModule::get_int(boma, *FIGHTER_INSTANCE_WORK_ID_INT_ENTRY_ID) as usize;
+				let frame = MotionModule::frame(boma);
+				if !smash::app::sv_information::is_ready_go(){
+					VarModule::set_int(fighter.battle_object, vars::rosetta::instance::COOLDOWN, 0);
+					VarModule::off_flag(boma.object(), vars::rosetta::instance::IS_TICO_DEAD);
+				};
+				VarModule::set_int(fighter.battle_object, vars::rosetta::instance::TICO_RAYCAST, 
```

Comment:
I would appreciate that you don't nest VarModule calls for readability purposes.
`let rosa_x = ...;`, `let tico_x = ...;`, etc.